browser/brave_vpn/android/brave_vpn_native_worker.cc

```diff
@@ -0,0 +1,158 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_vpn/android/brave_vpn_native_worker.h"
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_array.h"
+#include "base/android/jni_string.h"
+#include "base/bind.h"
+#include "base/json/json_writer.h"
+#include "base/values.h"
+#include "brave/browser/brave_vpn/brave_vpn_service_factory.h"
+#include "brave/build/android/jni_headers/BraveVpnNativeWorker_jni.h"
+#include "brave/components/brave_vpn/brave_vpn_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+
+namespace chrome {
+namespace android {
+
+BraveVpnNativeWorker::BraveVpnNativeWorker(JNIEnv* env,
+                                 const base::android::JavaRef<jobject>& obj)
+    : weak_java_brave_vpn_native_worker_(env, obj),
+      brave_vpn_service_(nullptr),
+      weak_factory_(this) {
+  Java_BraveVpnNativeWorker_setNativePtr(env, obj, reinterpret_cast<intptr_t>(this));
+  brave_vpn_service_ = BraveVpnServiceFactory::GetForProfile(
```

Comment:
AFAIU this doesn't track the service runtime? the service can die at any moment, so we'd better ensure that we don't use this pointer after this.

Reply:
I think this wasn't fixed (or I don't understand why the current code is correct). Can you please clarify the lifetime of the worker? I.e. when it is created and when destroyed

Reply:
So I'd like to understand whether the worker can outlive the service

Reply:
This concern has been addressed.